mono/metadata/threads-types.h

```diff
@@ -216,7 +222,7 @@ gunichar2* mono_thread_get_name (MonoInternalThread *this_obj, guint32 *name_len
 MONO_API MonoException* mono_thread_get_undeniable_exception (void);
 void mono_thread_self_abort (void);
 
-void mono_thread_set_name_internal (MonoInternalThread *this_obj, MonoString *name, gboolean permanent, gboolean reset, MonoError *error);
+void mono_thread_set_name_internal (MonoInternalThread *this_obj, const char *name_utf8, size_t length, gboolean permanent, gboolean reset, MonoError *error);
```

Comment:
Why pass `length` when `name_utf8` is NULL-terminated making it possible to call `strlen` inside `mono_thread_set_name_internal`?

And we use `gsize` instead of `size_t`.

Reply:
Strlen is slow and wasteful and unnecessary. Avoid it. Size_t is really ancient and seems right, unless we insist on 32bit limits which I'd rather not. But stye/name ok.

Reply:
Does Text.Encoding even add the nul? That'd be kinda wierd I think, but maybe. The nul terminated string really is a mistake. A widespread one, I understand.

Reply:
No, good, System.TextEncoding does not add a terminal nul. I would not expect it to.

```
$ cat 1.cs
public class A
{
 public static void Main()
 {
 System.Console.WriteLine("A");
 byte[] name_utf8 = System.Text.Encoding.UTF8.GetBytes("A");
 System.Console.WriteLine("{0}", name_utf8.Length);
 }
}$ mcs 1.cs
$ mono ./1.exe
A
1
```

Granted, we could add another layer and call strlen on the C
literal, and take the length from managed, but this would seem
counter productive.

There is funny business I did not acknowledge

Reply:
Is it possible, there is a guaranteed zero at the end of the byte[], not represented in its .Length? That seems unlikely, but possible.

Reply:
There is therefore a bug where we pass non-null terminated to native and profiler. I'll fix.